src/clang-c-frontend/clang_c_adjust_code.cpp

```diff
@@ -57,7 +57,8 @@ void clang_c_adjust::adjust_code(codet &code)
        * unused.
        */
       exprt &op = code.op0();
-      op = address_of_exprt(index_exprt(op, constant_exprt(0, index_type())));
+      if (op.has_operands() && op.op0().type().is_pointer())
+        op = address_of_exprt(index_exprt(op, constant_exprt(0, index_type())));
```

Comment:
Doing this only for dereferenced pointers is the wrong condition. The point of this transformation is to avoid array-typed top-level expressions in statements. It's better to test whether the statement is actually an assignment. Because in that case we don't need to do the conversion. E.g., when just testing for pointer type, the following variation of #1938 starts failing again (it's fine on current master):
```c
struct S { int ss[128]; };

int main()
{
	struct S s, *t = &s;
	(*t).ss;
}
```